### PR TITLE
Add UUIDs module

### PR DESCRIPTION
**Pull request by iamed2**
These were available but unexported from Base.Random, then available but unexported from Random, then exported from UUIDs. They are included here unexported.

### CODE REVIEW

File: src/Compat.jl [hunk 1]
```diff
@@ -1629,6 +1629,12 @@ else
     findall(b::OccursIn, a::Number) = a in b.x ? [1] : Vector{Int}()
 end
 
+if VERSION >= v"0.7.0-DEV.3666"
+    using UUIDs: uuid4
+else
+    using .Random: uuid4
```
Comment (martinholters) [inline src/Compat.jl]: Shouldn't these include `uuid1` and `uuid_version`?

**Comment by martinholters** [comment]
```
ERROR: LoadError: ArgumentError: Module UUIDs not found in current path.
Run `Pkg.add("UUIDs")` to install the UUIDs package.
```
Do we have to add an entry to `Project.toml` now?

**Comment by iamed2** [comment]
Now it makes a module and all review comments should be resolved

**Comment by martinholters** [comment]
Still no tests?

**Comment by iamed2** [comment]
There are tests now

File: src/Compat.jl [hunk 2]
```diff
@@ -1629,6 +1629,15 @@ else
     findall(b::OccursIn, a::Number) = a in b.x ? [1] : Vector{Int}()
 end
 
+if VERSION >= v"0.7.0-DEV.3666"
+    import UUIDs
+else
+    @eval module UUIDs
+        import .Random: uuid1, uuid4, uuid_version
```
Comment (martinholters) [inline src/Compat.jl]:
```
ERROR: LoadError: in module path: Random not defined
Stacktrace:
 [1] eval(::Module, ::Expr) at ./sysimg.jl:23
 [2] include_from_node1(::String) at ./loading.jl:576
 [3] include(::String) at ./sysimg.jl:14
 [4] anonymous at ./<missing>:2
while loading /home/travis/.julia/v0.6/Compat/src/Compat.jl, in expression starting on line 1635
ERROR: LoadError: Failed to precompile Compat to /home/travis/.julia/lib/v0.6/Compat.ji.
```

I think this has to be `..Random`.

Also, `UUIDs` on Julia 0.7 exports `UUID`, so I guess it makes sense to include it here.

File: test/runtests.jl [hunk 3]
```diff
@@ -1393,4 +1393,14 @@ end
...
 @test findall(occursin([1, 2]), [1]) == [1]
 
+# 0.7.0-DEV.3666
+module TestUUIDs
+    using Compat
+    using Compat.UUIDs
+    using Compat.Test
+    @test isdefined(@__MODULE__, :uuid1)
+    @test isdefined(@__MODULE__, :uuid4)
+    @test isdefined(@__MODULE__, :uuid_version)
```
Comment (martinholters) [inline test/runtests.jl]: Eh, should test the behavior, too (to the extent possible with randomness). E.g. `uuid_version(uuid1()) == 1` and `uuid1() isa UUID`, similar for `uuid4`.

Comment (iamed2) [reply test/runtests.jl]: Added

File: src/Compat.jl (hunk 2 above)
Comment (iamed2) [reply src/Compat.jl]: Fixed/added

**Comment by martinholters** [comment]
Thanks! Remaining CI failures look unrelated; will merge later unless objections are voiced.

File: README.md [hunk 4]
```diff
@@ -552,4 +555,4 @@ includes this fix. Find the minimum version from there.
 [#25706]: https://github.com/JuliaLang/julia/issues/25706
 [#25780]: https://github.com/JuliaLang/julia/issues/25780
 [#24182]: https://github.com/JuliaLang/julia/issues/24182
-[#24673]: https://github.com/JuliaLang/julia/issues/24673
\ No newline at end of file
+[#24673]: https://github.com/JuliaLang/julia/issues/24673
```
Comment (omus) [inline README.md]: Doesn't match the PR number above

Comment (fredrikekre) [reply README.md]: ran the script in #493

**Pull request merged**